src/frontend/src/binder/select.rs

```diff
@@ -435,6 +508,61 @@ impl Binder {
             }
         })
     }
+
+    /// Uses a literal value to look up the ID of an Object (e.g. a table). If the literal is an
+    /// integer (int16, int32, or int64) this will just return that number as an ObjectID value.
+    /// If the literal is a varchar, this will look in the Catalog for an object with a name that
+    /// matches the varchar and return its Object ID value; of no match is found then an error is
+    /// returned.
+    fn bind_object_id(&mut self, arg: &Literal) -> Result<TableId> {
+        match arg
+            .get_data()
+            .as_ref()
+            .ok_or_else(|| ErrorCode::BindError("No Value".to_string()))?
+        {
+            ScalarImpl::Int16(id) => Ok(TableId::new(*id as u32)),
+            ScalarImpl::Int32(id) => Ok(TableId::new(*id as u32)),
+            ScalarImpl::Int64(id) => Ok(TableId::new(*id as u32)),
+            ScalarImpl::Utf8(name) => {
+                let object = self.get_object_by_name(name)?;
+                Ok(object.table_id)
+            }
+            _ => Err(ErrorCode::BindError(
+                "This only supports Object IDs (int) or Object Names (varchar) literals."
+                    .to_string(),
+            ))?,
+        }
+    }
+
+    /// Attempt to get the reference to a Database object by it's name.
+    fn get_object_by_name(&mut self, name: &str) -> Result<BoundBaseTable> {
+        let object_name = Self::parse_object_name(name)?;
+        let (schema_name, table_name) =
+            Self::resolve_schema_qualified_name(&self.db_name, object_name)?;
+
+        self.bind_table(schema_name.as_deref(), &table_name, None)
```

Comment:
Should I use bind the table whose size is being looked up? I did this because using bind_table was the existing on Binder that would provide the TableId needed to query rw_table_stats. But binding the table from the pg_table_size argument strikes me as less than safe and, instead, I should use a function that looks up the Table information but does not bind the table to the query compiler context.